Code/Framework/AzToolsFramework/AzToolsFramework/AssetBrowser/Views/EntryDelegate.cpp

```diff
@@ -148,7 +154,163 @@ namespace AzToolsFramework
             return m_iconSize;
         }
 
-    } // namespace Thumbnailer
+        SearchEntryDelegate::SearchEntryDelegate(QWidget* parent)
+            : EntryDelegate(parent)
+        {
+            LoadBranchPixMaps();
+        }
+
+        void SearchEntryDelegate::paint(QPainter* painter, const QStyleOptionViewItem& option, const QModelIndex& index) const
+        {
+            auto data = index.data(AssetBrowserModel::Roles::EntryRole);
+            if (data.canConvert<const AssetBrowserEntry*>())
+            {
+                bool isEnabled = (option.state & QStyle::State_Enabled) != 0;
+                bool isSelected = (option.state & QStyle::State_Selected) != 0;
+
+                QStyle* style = option.widget ? option.widget->style() : QApplication::style();
+
+                // draw the background
+                style->drawPrimitive(QStyle::PE_PanelItemViewItem, &option, painter, option.widget);
+
+                // Draw main entry thumbnail.
+                QRect remainingRect(option.rect);
+
+                QSize iconSize(m_iconSize, m_iconSize);
+                // Note that the thumbnail might actually be smaller than the row if theres a lot of padding or font size
+                // so it needs to center vertically with padding in that case:
+                QPoint iconTopLeft; //(remainingRect.x(), remainingRect.y() + (remainingRect.height() / 2) - (m_iconSize / 2));
+                QPoint branchIconTopLeft = QPoint();
```

Comment:
question: What happens if this is used later if it does not enter the else block lower down? Is it fine if it's zero?

Reply:
Yes that is totally find because no icon will be loaded in that case